src/actions/collapseContext.ts

```diff
@@ -76,17 +83,52 @@ const collapseContext = (state: State, { at }: Options) => {
       newValue: createId(), // unique value
       path: simplePath,
     }),
-    // Sort parent context if sort preference exists
+    // Sort parent context if sort preference collapses
     sortId ? sort(parentId) : null,
+
     // outdent each child
-    ...children.map(
-      (child, i) => (state: State) =>
-        moveThought(state, {
+    ...children.map(child => (state: State) => {
+      // If we're inserting into a sorted context, short-circuit and use the sorted rank
+      if (isSortedContext)
+        return moveThought(state, {
           oldPath: appendToPath(simplePath, child.id),
           newPath: appendToPath(parentOf(simplePath), child.id),
-          newRank: sortId ? getSortedRank(state, parentId, child.value) : rankStart + rankIncrement * i,
-        }),
-    ),
+          newRank: getSortedRank(state, parentId, child.value),
+        })
```

Comment:
There are two minor but justifiable reasons to structure this conditional logic differently:

1. `oldPath` and `newPath` are identical in both cases and end up getting duplicated in the code. The developer has to look back and forth to confirm that they are indeed the same.
2. A single return statement is preferable over multiple return statements within a function. Knowing that there is a single exit point at the end of the block makes it easier to reason about.

Therefore, I recommend just using the `isSortedContext` condition to set `let newRank` and have a single return statement at the end.